feat(GiniHealthSDK): Payment component documentation and integration guide IPC-98 (#453)
gini/gini-mobile-ios: HealthSDK/GiniHealthSDK/Documentation/source/Integration.md

.
healthSDK.setConfiguration(config)
```

## Payment component view initialization
We provide a custom payment component view meant to help user to pay the invoice/document.
Payment component looks like this: 
<br>
<center><img src="img/Customization guide/PaymentComponentView.PNG" height="400"/></center>
</br>

##### Steps to integrate payment component view:

##### 1. You need to instantiate a `PaymentComponentsController` class with a `GiniHealth` as parameter.
```swift
let paymentComponentsController = PaymentComponentsController(giniHealth: health)
```

##### 2. You need to pass this `paymentComponentsController` to the invoices/documents controller. 

> There you should have a list of invoices/documents that conform to this structure: `GiniDocument`. 

Reviewer: [suggestion]: I would rewrite this like this:
There you should have a list of invoices/documents that conform to`GiniDocument` protocol.
How does this sound?

    /// The document's unique identifier.
    var documentID: String { get set }
    /// The document's amount to pay.
    var amountToPay: String? { get set }
    /// The document's payment due date.
    var paymentDueDate: String? { get set }
    /// The document's recipient name.
    var recipient: String? { get set }
    /// Boolean value that indicates if the document is payable. This is obtained by calling the checkIfDocumentIsPayable method.
    var isPayable: Bool? { get set }
    /// Store payment provider for each document/invoice. Payment provider is obtained by PaymentComponentsController and it's only stored here if the payment provider is installed
    var paymentProvider: PaymentProvider? { get set }
}
```

##### 3. For each invoice/document, you need to call the `checkIfDocumentIsPayable` function from the `paymentComponentsController` and store the value for each invoice in the `isPayable` field.
> - We recommend using a `DispatchGroup` for these requests and wait till all of them are ready. Then, reload the list.
```swift
public func checkIfDocumentIsPayable(docId: String, completion: @escaping (Result<Bool, GiniHealthError>) -> Void)
```

> ##### 4. Also, after `paymentComponentsController` is instantiated, he will load in the background the list of payment providers available on your device. You should store for each invoice/document, in the `paymentProvider` variable, the first payment provider from the list. 

Reviewer: [clarification]: who is he in this case: `Also, after `paymentComponentsController` is instantiated, he will load in the background the list of payment providers available on your device.` ?

Is not better to rewrite something like this:

_Also, after `paymentComponentsController` is instantiated, the list of payment providers available on your device will be loaded in the background._

> You can obtain that by calling: 

Reviewer: [suggestion]: I suggest using `achieve` instead of obtain. How does it sound?

```swift
public func obtainFirstPaymentProvider() -> PaymentProvider?
```

> > - You might have payment providers, or you may not. We also handle the missing payment provider state.

Reviewer: [suggestion]: `We effectively manage situations where there are no payment providers available.`
Any thoughts?

> - Based on the payment provider's colors, the `UIView` will automatically change it's color.

> ##### 5. Based on the `isPayable` value, you should add to your cells the payment component view that is obtained through this function:

Reviewer: [suggestion:] Does this sound better:
Depending on the value of `isPayable`, incorporate the corresponding payment component view into your cells using this function:

```swift
public func getPaymentView(paymentProvider: PaymentProvider?) -> UIView
```

> > - We recommend adding this `UIView` to a Vertical `UIStackView` and in every cell's `prepareForReuse()` function, remove the payment component view if it's present.

Reviewer: [suggestion]: How this is sounds:

We suggest placing this `UIView` within a vertical `UIStackView`. Additionally, in the `prepareForReuse()` function of each cell, remove the payment component view if it exists. 
Furthermore, employing automatic dimension height in the `UITableView` containing the cells is recommended.

Payment component looks like this: 
<br>
<center><img src="img/Customization guide/PaymentComponentView.PNG" height="400"/></center>
</br>

### Steps to integrate payment component view:

#### 1. You need to create an instance of the `PaymentComponentsController` class with a `GiniHealth` as parameter.
```swift
let paymentComponentsController = PaymentComponentsController(giniHealth: health)
```

#### 2. You need to pass this `paymentComponentsController` to the invoices/documents controller. 
There you should have a list of invoices/documents that conform toGiniDocument protocol.
```swift
public protocol GiniDocument {
    /// The document's unique identifier.
    var documentID: String { get set }
    /// The document's amount to pay.
    var amountToPay: String? { get set }
    /// The document's payment due date.
    var paymentDueDate: String? { get set }
    /// The document's recipient name.
    var recipient: String? { get set }
    /// Boolean value that indicates if the document is payable. This is obtained by calling the checkIfDocumentIsPayable method.
    var isPayable: Bool { get set }

>     /// Stored payment provider for each document/invoice. Payment provider is obtained by PaymentComponentsController and it's only stored here if the payment provider is installed

Author: Payment provider should be stored when you call `obtainFirstInstalledPaymentProvider` method when you add a new invoice to the list or when list of available payment providers is fetched on `didFetchedPaymentProviders`.

>     /// Stored payment provider for each document/invoice. Payment provider is obtained by PaymentComponentsController and it's only stored here if the payment provider is installed

Author: We are no longer storing the payment provider in here.

       }
       self?.dispatchGroup.leave()
   })
}
dispatchGroup.notify(queue: .main) {
    // Reload List
}
```

#### 4. Also, after `paymentComponentsController` is instantiated, the list of payment providers available on your device will be loaded in the background. Store for each invoice/document, in the `paymentProvider` variable, the first payment provider from the list. 
That can be achieved by calling:
```swift
public func obtainFirstPaymentProvider() -> PaymentProvider?
```
> - We effectively manage situations where there are no payment providers available.
> - Based on the payment provider's colors, the `UIView` will automatically change it's color.

#### 5. Depending on the value of `isPayable`, incorporate the corresponding payment component view into your cells using this function:
```swift
public func getPaymentView(paymentProvider: PaymentProvider?) -> UIView
```
> - We suggest placing this `UIView` within a vertical `UIStackView`. Additionally, in the `prepareForReuse()` function of each cell, remove the payment component view if it exists.
> - Furthermore, employing automatic dimension height in the `UITableView` containing the cells is recommended.

> #### 6. `PaymentComponentsController` has 2 delegates that you can listen to: `PaymentComponentsControllerProtocol` and the `PaymentComponentViewProtocol`

Author: No, the user should listen to both in order to have a complete flow.

* `PaymentComponentsControllerProtocol` provides informations when the `PaymentComponentsController` is loading. You can show/hide an `UIActivityIndicator` based on that.
* `PaymentComponentsControllerProtocol` provides completion handlers when `PaymentComponentsController` fetched successfully payment providers or when it failed with an error.

> * `PaymentComponentViewProtocol` is the view protocol and provides events when the user tapped on various areas on the payment component view. Such as: More information label or more information icon, bank/payment provider picker or when it tapped on the pay invoice button.

Reviewer 2: ```suggestion
* `PaymentComponentViewProtocol` is the view protocol and provides events handlers when the user tapped on various areas on the payment component view (more information icon, bank/payment provider picker, the pay invoice button and etc.).
```
